pkg/curatedpackages/curatedpackages.go

```diff
@@ -0,0 +1,61 @@
+package curatedpackages
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/go-logr/logr"
+
+	"github.com/aws/eks-anywhere-packages/pkg/artifacts"
+	"github.com/aws/eks-anywhere-packages/pkg/bundle"
+	"github.com/aws/eks-anywhere/pkg/dependencies"
+	"github.com/aws/eks-anywhere/pkg/helm"
+	"github.com/aws/eks-anywhere/pkg/version"
+)
+
+func NewRegistry(ctx context.Context, deps *dependencies.Dependencies, registry, kubeVersion string) (BundleRegistry, error) {
+	if registry != "" {
+		username, password, err := helm.ReadRegistryCredentials()
+		if err != nil {
+			return nil, err
+		}
+		registry := NewCustomRegistry(
+			deps.Helm,
+			registry,
+			username,
+			password,
+		)
+		err = registry.Login(ctx)
```

Comment:
Why don't we leave this until it is needed? Instead of doing it in the constructor (which is a bit unexpected) just call it from your "action" methods whenever is needed. 

Or you could add an `Init` method to `BundleRegistry` and make sure the callers invoke it before anything else.